File: app/src/main/java/com/hamboomger/model/event/EventAgenda.java
package com.hamboomger.model.event;

import javax.persistence.Access;
import javax.persistence.ElementCollection;
import javax.persistence.Embeddable;
import java.util.List;

/**
 * @author ddorochov
 */
@Embeddable
@Access(javax.persistence.AccessType.FIELD)
public class EventAgenda {

	@ElementCollection
	private List<String> appointments;

	public EventAgenda(List<String> appointments) {
		this.appointments = appointments;
	}

	protected EventAgenda() {}

	public List<String> getAppointments() {
		return appointments;
	}

}

File: app/src/main/java/com/hamboomger/model/event/EventAgenda.kt
package com.hamboomger.model.event

import java.time.LocalTime

/**
 * @author ddorochov
 */
class EventAgenda private constructor(builder: Builder) {
    val appointments : List<Appointment>
    val additionalInfo : String

    class Builder {
        val appointments = mutableListOf<Appointment>()
        val additionalInfo = StringBuilder()

        fun addAppointment(startTime: LocalTime, description: String) {
            appointments.add(Appointment(startTime, description))
        }

        fun addAdditionalInfo(info : String) {
            additionalInfo.appendln(info)
        }

        fun build() : EventAgenda = EventAgenda(this)
    }

    init {
        this.appointments = builder.appointments
        this.additionalInfo = builder.additionalInfo.toString()
    }

}

class Appointment(val startTime : LocalTime, val description: String)